### PR TITLE
Fixes #9089 - refactored SSL client verification into method

### PR DESCRIPTION
**Pull request by lzap**
Proper fix instead of the
https://github.com/theforeman/smart_proxy_discovery/pull/4

### CODE REVIEW

**Comment by dmitri-d** [comment]
this looks good, although a bunch of tests failed.

**Comment by domcleal** [comment]
Please also ensure all plugins currently available get this added to them where necessary to avoid a regression.

**Comment by lzap** [comment]
I only see foreman_reverse_proxy in our list. Do you have something else in mind?

http://projects.theforeman.org/projects/foreman/wiki/List_of_Plugins

**Comment by domcleal** [comment]
http://projects.theforeman.org/projects/foreman/wiki/List_of_Smart-Proxy_Plugins

**Comment by lzap** [comment]
Fixed the unit tests. Now the question - can we accept this into 1.7.3? It's a quite a change which can possible prevent some plugins from working until their authors release a fix. While I can send out all the PRs easily (I will), this can be bummer for users.

**Comment by domcleal** [comment]
No chance.. removing security features by default isn't something that would go into a patch release near EOL..

**Comment by lzap** [comment]
No big deal since 1.8 is pretty close.

**Comment by lzap** [comment]
Will file PRs for the plugins once this is merged.

File: modules/root/root_api.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class Proxy::RootApi < Sinatra::Base
   helpers ::Proxy::Helpers
   authorize_with_trusted_hosts
+  authorize_with_ssl_client
```
Comment (dmitri-d) [inline modules/root/root_api.rb]: I don't think we need any authorization (including trusted hosts) here. The only available data in the calls in this module is proxy features and version.

Comment (lzap) [inline modules/root/root_api.rb]: Removed.

**Comment by dmitri-d** [comment]
merged in at 0207401. Thanks, @lzap!

**Comment by lzap** [comment]
Thanks, working on those patches for plugins. Feel free to help.

**Comment by lzap** [comment]
Filed PR for ABRT: https://github.com/theforeman/smart_proxy_abrt/pull/3

**Comment by lzap** [comment]
Filed PR for Pulp: https://github.com/theforeman/smart-proxy-pulp/pull/2

**Comment by lzap** [comment]
Filed PR for OpenSCAP: https://github.com/OpenSCAP/smart_proxy_openscap/pull/9

**Comment by lzap** [comment]
Filed PR for Chef: https://github.com/theforeman/smart_proxy_chef/pull/7

**Comment by lzap** [comment]
For the record Discovery plugin does not need this change.

**Comment by lzap** [comment]
Filed PR for Salt: https://github.com/theforeman/smart_proxy_salt/pull/8

That's all folks.

**Pull request closed without merging**